tests/units/test_runner.py

```diff
@@ -7,59 +7,44 @@
 
 import logging
 import resource
+import sys
 from pathlib import Path
 from unittest.mock import patch
 
 import pytest
 
-from anta import logger
 from anta.catalog import AntaCatalog
 from anta.inventory import AntaInventory
 from anta.result_manager import ResultManager
 from anta.runner import adjust_rlimit_nofile, main, prepare_tests
 
-from .test_models import FakeTest
+from .test_models import FakeTest, FakeTestWithMissingTest
 
 DATA_DIR: Path = Path(__file__).parent.parent.resolve() / "data"
 FAKE_CATALOG: AntaCatalog = AntaCatalog.from_list([(FakeTest, None)])
 
 
-async def test_runner_empty_tests(caplog: pytest.LogCaptureFixture, test_inventory: AntaInventory) -> None:
-    """Test that when the list of tests is empty, a log is raised.
-
-    caplog is the pytest fixture to capture logs
-    test_inventory is a fixture that gives a default inventory for tests
-    """
-    logger.setup_logging(logger.Log.INFO)
+async def test_empty_tests(caplog: pytest.LogCaptureFixture, inventory: AntaInventory) -> None:
+    """Test that when the list of tests is empty, a log is raised."""
     caplog.set_level(logging.INFO)
     manager = ResultManager()
-    await main(manager, test_inventory, AntaCatalog())
+    await main(manager, inventory, AntaCatalog())
 
     assert len(caplog.record_tuples) == 1
     assert "The list of tests is empty, exiting" in caplog.records[0].message
 
 
-async def test_runner_empty_inventory(caplog: pytest.LogCaptureFixture) -> None:
-    """Test that when the Inventory is empty, a log is raised.
-
-    caplog is the pytest fixture to capture logs
-    """
-    logger.setup_logging(logger.Log.INFO)
+async def test_empty_inventory(caplog: pytest.LogCaptureFixture) -> None:
+    """Test that when the Inventory is empty, a log is raised."""
     caplog.set_level(logging.INFO)
     manager = ResultManager()
-    inventory = AntaInventory()
-    await main(manager, inventory, FAKE_CATALOG)
+    await main(manager, AntaInventory(), FAKE_CATALOG)
     assert len(caplog.record_tuples) == 3
     assert "The inventory is empty, exiting" in caplog.records[1].message
 
 
-async def test_runner_no_selected_device(caplog: pytest.LogCaptureFixture, test_inventory: AntaInventory) -> None:
-    """Test that when the list of established device.
-
-    caplog is the pytest fixture to capture logs
-    test_inventory is a fixture that gives a default inventory for tests
-    """
-    logger.setup_logging(logger.Log.INFO)
+async def test_no_selected_device(caplog: pytest.LogCaptureFixture, test_inventory: AntaInventory) -> None:
+    """Test that when the list of established device."""
```

Comment:
hmm this sentence was never finished -looks like me
```suggestion
    """Test that when the list of established devices is empty a log is raised."""
```